src/classes/PMT_Payment_TDTM.cls

```diff
@@ -201,4 +162,132 @@ public class PMT_Payment_TDTM extends TDTM_Runnable {
+    * @return void              Does not return anything
+    ********************************************************************************************************/
+    private void setPaymentCurrency(Map<Id, Opportunity> mapOfOpps, npe01__OppPayment__c pmt, String oldPmtCurrency, TDTM_Runnable.Action triggerAction) {
+        String oppCurrency = (String) mapOfOpps.get(pmt.npe01__Opportunity__c).get('CurrencyIsoCode');
+        String pmtCurrency = (String) pmt.get('CurrencyIsoCode');
+
+        // Prevent the Payment CurrencyCode from being modified directly
+        if (triggerAction == TDTM_Runnable.Action.BeforeUpdate && pmtCurrency != oldPmtCurrency && pmtCurrency != oppCurrency) {
+            pmt.addError(System.Label.pmtModifyCurrency);
+
+        } else if (triggerAction == TDTM_Runnable.Action.BeforeInsert && pmtCurrency != oppCurrency && pmtCurrency != null) {
+            // If the User is attempting to set the Payment Currency to something different than the Opp currency
+            pmt.addError(System.Label.pmtModifyCurrency);
+
+        } else if (triggerAction == TDTM_Runnable.Action.BeforeInsert) {
+            // Otherwise default the Payment CurrencyIsoCode to the value from the parent Opportunity
+            pmt.put('CurrencyIsoCode', oppCurrency);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description              For multicurrency orgs with advanced currency management enabled, 
+    *                           Set the stageName if the amount is equal payments made and opp is not closed
+    * @param opp                Opportunity object
+    * @param closedStageName    String of the stageName
+    * @param dmlWrapper         DmlWrapper Object
+    * @return void              Does not return anything
+    ********************************************************************************************************/
+    private void handleAdvancedCurrencyPayments(Opportunity opp, String closedStageName, DmlWrapper dmlWrapper) {
+        Decimal paidAmount = PMT_PaymentCreator.getPaymentsMade(opp);
+
+        if (opp.Amount > 0 && paidAmount == opp.Amount && !opp.isClosed && !opp.IsWon) {
+            opp.StageName = closedStageName;
+            dmlWrapper.objectsToUpdate.add(opp);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description          Validate that the payment has the appropriate values
+    * @param newlist        The list of Payments from trigger new.
+    * @param oldlist        The list of Payments from trigger old.
+    * @param triggerAction  Which trigger event (BeforeInsert, AfterInsert, etc.).
+    * @return void          Does not return anything
+    ********************************************************************************************************/
+    private void validatePayment(
```

Comment:
Since this validates all the payment records, should the method be `validatePayments`